### DISCUSSION TITLE
Logging path bug?

### DISCUSSION THREAD

**Discussion by JoeFox82** [nextgenhealthcare/connect#5869] [General]
Hi together,

today i installed Mirth Conenct on a Windows Server. In the install dialog I configured to use e:\log as log path. After starting Mirth Connect, log files are saved in the install path in C:\Program Files\Mirth Conenct\logs.

I looked at the log4j2.properties and the parameter _dir.logs_ is _E:\\\\log_ but the _appender.rolling.fileName_ and _appender.rolling.filePattern_ are still beginning with **logs**\...

```
# file appender
dir.logs = E:\\\\log
appender.rolling.type = RollingFile
appender.rolling.name = fout
appender.rolling.fileName = logs/mirth.log
appender.rolling.filePattern = logs/mirth.log.%i

```

Because the _appender.rolling.fileName_ and _appender.rolling.filePattern_ are relative paths the mirth server could not find the dir.logs path and did a fallback to the root path of Mirth Connect.
After setting the dir.logs path to E:\\\\**logs** and restarting mirth, all log files are in E:\\\\logs.

Is this a bug or just my failure and the log path has always to be named _logs_.

Kind regards,
Jochen

**Comment by jonbartels** [comment]
Mirth Connect uses Log4J so its logging is entirely done with the `log4j2.properties` file. Mirth does not do anything special with logs, it is all Log4J. 

The Log4J config information is here: https://logging.apache.org/log4j/2.x/manual/configuration.html

I think what is happening is that the rolling file appender has to create multiple files so it can rotate its own logs. So it would be configured with a directory for the logs, then the relative paths from that directory to handle its rotation rules.

**Reply by JoeFox82** [reply]
When installing Mirth Connect the setup changes the parameter _dir.logs_ based on the user input. Perhaps it is possible, that the _appender.rolling.fileName_ and _appender.rolling.filePattern_ are changed automatically like this, too:

`appender.rolling.fileName = ${dir.logs}/mirth.log`
`appender.rolling.filePattern = ${dir.logs}/mirth.log.%i`

**Reply by jonbartels** [reply]
https://logging.apache.org/log4j/2.x/manual/configuration.html#property-substitution
